test/index.ts

```diff
@@ -251,6 +251,56 @@ function testDeepPartial() {
   ];
 }
 
+function testDeepNullable() {
+  type cases = [
+    Assert<IsExact<DeepNullable<number>, number | null>>,
+    Assert<IsExact<DeepNullable<string>, string | null>>,
+    Assert<IsExact<DeepNullable<boolean>, boolean | null>>,
+    Assert<IsExact<DeepNullable<bigint>, bigint | null>>,
+    Assert<IsExact<DeepNullable<symbol>, symbol | null>>,
+    Assert<IsExact<DeepNullable<undefined>, undefined | null>>,
+    Assert<IsExact<DeepNullable<null>, null>>,
+    Assert<IsExact<DeepNullable<Function>, Function | null>>,
+    Assert<IsExact<DeepNullable<Date>, Date | null>>,
+    Assert<IsExact<DeepNullable<Error>, Error | null>>,
+    Assert<IsExact<DeepNullable<RegExp>, RegExp | null>>,
+    Assert<IsExact<DeepNullable<Map<string, boolean>>, Map<string | null, boolean | null>>>,
+    Assert<IsExact<DeepNullable<Map<string, { a: number }>>, Map<string | null, { a: number | null }>>>,
```

Comment:
I am not sure if the behavior to make **keys** of Maps etc, `nullable` is desired. Have you thought about this?

Reply:
It's not a change, that's an alignment for `ReadonlyMap`, because it was already working that way for `Map` and `WeekMap`

Same regarding `Set` and `WeekSet` – it was already working this way and I aligned it for `ReadonlySet`

Reply:
Agreed on adding it as an alignment ✅